pinot-segment-local/src/test/java/org/apache/pinot/segment/local/utils/nativefst/mutablefst/MutableFSTConcurrentTest.java

```
@@ -102,8 +100,54 @@ public void testConcurrentWriteAndRead()
     assertTrue("efg not found in result set", _resultSet.contains("efg"));
   }
 
-  private void performReads(MutableFST fst, List<String> words, int count,
-      long sleepTime)
+  @Test
+  public void testConcurrentLongWriteAndRead()
+      throws InterruptedException {
+    MutableFST mutableFST = new MutableFSTImpl();
+    List<String> words = new ArrayList<>();
+    CountDownLatch countDownLatch = new CountDownLatch(2);
+
+    mutableFST.addPath("ab", 1);
+
+    List<Pair<String, Integer>> wordsWithMetadata = new ArrayList<>();
+
+    // Add some write pressure
+    wordsWithMetadata.add(Pair.of("egegdgrbsbrsegzgzegzegegjntnmtj", 2));
+    wordsWithMetadata.add(Pair.of("hrwbwefweg4wreghrtbrassregfesfefefefzew4ere", 2));
+    wordsWithMetadata.add(Pair.of("easzegfegrertegbxzzez3erfezgzeddzdewstfefed", 2));
+    wordsWithMetadata.add(Pair.of("tjntrhndsrsgezgrsxzetgteszetgezfzezedrefzdzdzdzdz", 2));
+    wordsWithMetadata.add(Pair.of("abacxcvbnmlkjjhgfsaqwertyuioopzxcvbnmllkjshfgsfawieeiuefgeurfeoafa", 2));
+
+    words.add("abacxcvbnmlkjjhgfsaqwertyuioopzxcvbnmllkjshfgsfawieeiuefgeurfeoafa");
+
+    _threadPool.submit(() -> {
+      try {
+        performReads(mutableFST, words, 10, 10, countDownLatch);
+      } catch (InterruptedException e) {
+        e.printStackTrace();
```

Wrap it in `RuntimeException`

---

Please don't swallow the exception in the test. We want to catch them